ryven/main/Ryven.py

```diff
@@ -1,105 +1,559 @@
+        dest='example',
+        help='load an example project '
+             '(do not give PROJECT argument)')
+
+    # Display
+
+    group = parser.add_argument_group('display')
+
+    group.add_argument(
+        '-w', '--window-theme',
+        choices=['dark', 'light', 'plain'],
+        default='dark',
+        dest='window_theme',
+        help='set the window theme '
+             '(default: %(default)s)')
+
+    group.add_argument(
+        '-f', '--flow-theme',
+        choices=[
+            'toy', 'tron', 'ghost', 'blender', 'simple', 'ueli',
+            'pure dark', 'colorful dark',
+            'pure light', 'colorful light', 'industrial', 'fusion'],
+        dest='flow_theme',
+        help='set the theme of the flow view '
+             '(default: {pure dark|pure light}, depending on the window theme)')
+
+    group.add_argument(
+        '--performance',
+        choices=['pretty', 'fast'],
+        default='pretty',
+        dest='performance',
+        help='select performance mode '
+             '(default: %(default)s)')
+
+    # TODO: Python >= 3.9
+    # group.add_argument(
+    #     '--animations',
+    #     action=argparse.BooleanOptionalAction,
+    #     dest='animations',
+    #     help='en/disable animations '
+    #          '(default: %(default)s)')
+    exclusive_group = group.add_mutually_exclusive_group()
+    exclusive_group.add_argument(
+        '--no-animations',
+        action='store_false',
+        dest='animations',
+        help='do not use animations '
+             '(default: use animations)')
+    exclusive_group.add_argument(
+        '--animations',
+        action='store_true',
+        dest='animations',
+        help='use animations '
+             '(default: use animations)')
+
+    # TODO: Python >= 3.9
+    # group.add_argument(
+    #     '--info-messages',
+    #     action=argparse.BooleanOptionalAction,
+    #     dest='info_messages',
+    #     help='en/disable info messages '
+    #          '(default: %(default)s)')
+    exclusive_group = group.add_mutually_exclusive_group()
+    exclusive_group.add_argument(
+        '--info-messages',
+        action='store_true',
+        dest='info_messages',
+        help='show info messages '
+             '(default: do not show info messages')
+    exclusive_group.add_argument(
+        '--no-info-messages',
+        action='store_false',
+        dest='info_messages',
+        help='do not show info messages '
+             '(default: do not show info messages')
+
+    # Debug/Verbose
+
+    group = parser.add_argument_group('verbose')
+
+    group.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        dest='verbose',
+        help='display messages on the console')
+
+    # FIXME: Is there a way to get the version dynamically?
+    # Could also be used in `MainWindow.save_project()`
+    group.add_argument(
+        '-V', '--version',
+        action='version',
+        version='%(prog)s 3.1')
```

Comment:
I have seen many Python modules, which simply hard code the version number. This requires an updating of the version string with every release …

IMHO the better way would be to have a
```python3
__VERSION__ = '3.1'
```
string near the start of the file and then something like
```python3
    group.add_argument(
        '-V', '--version',
        action='version',
        version=f'%(prog)s {__VERSION__}')
```

Reply:
Indeed, that's what I have seen as well, I was just hoping to avoid this redundancy :(

Reply:
Yeah, I do not know, how often I forgot to update that version string!

Here are 6 ways, how to have one single source of truth: https://packaging.python.org/en/latest/guides/single-sourcing-package-version/ Which one do you prefer?

BTW, it should be `__version__` instead of `__VERSION__` and:
```python3
__version_info__ = (1, 2, 3)
__version__ = '.'.join([str(n) for n in __version_info__])
```

Reply:
Oh, I was just happy to see that option 1 shows how I can do this in compliance with `setup.cfg`. Until I discovered that it doesn't work at all. Older (but very common) versions of `setuptools` are not supposed to work, and I cannot get it to work with a newer version either (still having those import issues: during setup it imports, hence *executes* the file instead of ASTing it, which causes dependency `import`s to execute before dependencies are even parsed).

I now see two simple solutions for this:
1. hard code
2. put all available metadata with programmatic availability requirement into separate module (e.g. `pkg_info.py`) that does not cause imports of any sources

Option 2 is clean but kinda overkill for just the version, and would then be a mess because some metadata is in this file, and some other is in `setup.cfg`, so I think I would hard code it for now. What do you think?

Reply:
Personally I like option 4., but this might cause a slight delay when starting... But both hard coding or separate module are also perfectly ok with me.

Reply:
yes, option 4 would work when setting metadata attributes in `setup.py` but I'm doing it in `setup.cfg`, and reading from a text file is not even supported there, I think : /

Reply:
I implemented it hardcoded (in `setup.cfg` and `__init__.py`) for now, see e8125b45e301eabe34834fe4ce69fe54f83f8387

I'm happy to change it if there's a way to nicely load it from a single source.

Reply:
Aargh, I was not looking closely enough at the suffixes!! I was always thinking `setup.py` and not `.cfg`. Sorry for the confusion.

Reply:
ah I see, no worries! : )